src/include/rados/librados.hpp

```diff
@@ -1154,6 +1154,9 @@ namespace librados
 
     config_t cct();
 
+    void set_honor_osdmap_full();
+    void unset_honor_osdmap_full();
+
```

Comment:
Instead of exposing this, you can use the existing librados op flag:
<pre>
src/include/rados/librados.hpp:    OPERATION_FULL_TRY           = LIBRADOS_OPERATION_FULL_TRY,
</pre>
FULL_TRY makes the OSD assemble the transaction and proceed only if it does not result in a net increase in utilization.  This should be true for deleting objects.

There's also a FORCE variant, but I don't think you should need it...